OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java

```diff
@@ -407,4 +413,29 @@ public UnsavedWorkState canOpen() {
             return UnsavedWorkState.OPEN;
         }
     }
+
+    private void autoDiscoverLaTeXPath() {
+        try {
+            String command;
+
+            if (System.getProperty("os.name").contains("Windows")) {
+                command = "where " + ApplicationUtils.TEX_NAME;
+            } else {
+                command = "which " + ApplicationUtils.TEX_NAME;
+            }
+
+            Process p = Runtime.getRuntime().exec(command);

```

Comment:
In my opinion using Apache Commons is a little overkill for running such a small program. That's how it would look like: 
![apachecommonsexec](https://cloud.githubusercontent.com/assets/17130832/16945949/9af616b2-4da7-11e6-83f9-00bc3751ec00.png)
I had to cut off the last char (line 431), because it is the new line char. Anyway, this code works, but I tested it only on Windows.